src/org/labkey/test/components/list/AdvancedListSettingsDialog.java

```diff
@@ -0,0 +1,169 @@
+        return this;
+    }
+
+    public AdvancedListSettingsDialog allowOneDiscussionPerItem()
+    {
+        elementCache().radio(this, "Allow one discussion per item").check();
+        return this;
+    }
+
+    public AdvancedListSettingsDialog allowMultipleDiscussionsPerItem()
+    {
+        elementCache().radio(this, "Allow multiple discussions per item").check();
+        return this;
+    }
+
+    public AdvancedListSettingsDialog indexEntireListAsASingleDocument(boolean checked, String docTitle, String includeRadioLabel, String indexRadioLabel)
+    {
+        String labelText = "Index entire list as a single document";
+        elementCache().checkbox(labelText).set(checked);
+        if (checked)
+        {
+            WebElement expandContainer = elementCache().collapseibleFieldContainer(labelText)
+                    .waitForElement(this, 2000);
+            WebElement expandCollapsePane = elementCache().collapsibleField(labelText);
+            expandPane(expandCollapsePane);
+            Input.Input(Locator.id("entireListTitleTemplate"), getDriver()).find().set(docTitle);
+            elementCache().radio(expandContainer, includeRadioLabel).check();
+            elementCache().radio(expandContainer, indexRadioLabel).check();
+        }
+        return this;
+    }
+
+    public AdvancedListSettingsDialog indexEachItemAsASeparateDocument(boolean checked, String docTitle, String radioLabel)
+    {
+        String labelText = "Index each item as a separate document";
+        elementCache().checkbox(labelText).set(checked);
+        if (checked)
+        {
+            WebElement expandContainer = elementCache().collapseibleFieldContainer(labelText)
+                    .waitForElement(this, 2000);
+            WebElement expandCollapsePane = elementCache().collapsibleField(labelText);
+            expandPane(expandCollapsePane);
+
+            Input.Input(Locator.id("eachItemTitleTemplate"), getDriver()).find().set(docTitle);
+            elementCache().radio(expandContainer, radioLabel).check();
+        }
+        return this;
+    }
+
+    private void expandPane(WebElement expandCollapsePane)
+    {
+        if (!isPaneExpanded(expandCollapsePane))
+        {
+            expandCollapsePane.click();
+            WebDriverWrapper.waitFor(()-> isPaneExpanded(expandCollapsePane),
+                    "the pane did not expand in time", 1000);
+        }
+    }
+
+    private boolean isPaneExpanded(WebElement expandCollapsePane)
+    {
+        return Locator.tagWithClass("svg", "fa-angle-down").existsIn(expandCollapsePane);
+    }
+
+    public AdvancedListSettingsDialog setIndexFileAttachments(boolean checked)
+    {
+        Locator loc = Locator.tagWithClass("span", "list__advanced-settings-model__index-checkbox")
+                .withChild(Locator.tagWithText("span", "Index file attachments"))
+                .child(Locator.tagWithClass("span", "list__properties__checkbox--no-highlight"));
```

Comment:
I presume this predates the Locator in the element cache.
```suggestion
        Locator loc = elementCache().checkBoxLoc("Index file attachments");
```

Reply:
When I was testing this, it turned out that the span element on the checkboxes adjacent to expand-fields has a different class than standalone checkboxes- "model__index" vs "modal__index"
I just worked around this